src/ArtistInfo.cpp

```diff
@@ -0,0 +1,45 @@
+//
+// Created by shvmo on 03/05/2020.
+//
+
+#include "ArtistInfo.h"
+
+namespace DataStructuresWet1 {
+    DataStructuresWet1::ArtistInfo::ArtistInfo(int artistId, int numOfSongs) : artistId(artistId),
+                                                                               numOfSongs(numOfSongs) {
+        this->songs = unique_ptr<SongInfo[]>(new SongInfo[numOfSongs]);
+        this->songsPos = unique_ptr<shared_ptr<BiDirectionalNode<StreamCountGroup>>[]>(
+                new shared_ptr<BiDirectionalNode<StreamCountGroup>>[numOfSongs]);
+        this->songInfoListPointers = unique_ptr<shared_ptr<BiDirectionalNode<SongInfo>>[]>(
+                new shared_ptr<BiDirectionalNode<SongInfo>>[this->numOfSongs]);
```

Comment:
ok wtf? 😆

Reply:
hmmm can you extract some variables?

Reply:
Those variables are extracted... They're just templates so their Type is long

Reply:
It's kind of problemtic. I'm doing it on purpuse because I want the smart_ptrs to deal with allocation failure automatically. 
If I were to so all of the new allocations before calling the smart_ptr ctors we could have a potential memory leak. 
:(

Reply:
:( okay...